src/Main/Settings.h

```diff
@@ -1,39 +1,58 @@
-// sma find where to get current copyright text from
+/*
+Copyright 2022 The Foedag team
+
+GPL License
+
+Copyright (c) 2022 The Open-Source FPGA Foundation
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
 
 #ifndef SETTINGS_H
 #define SETTINGS_H
 
 #include <QJsonObject>
 #include <QJsonValue>
 
+#include "nlohmann_json/json.hpp"
+// Per https://json.nlohmann.me/features/object_order/
+// Json order is undefined in the JSON standard. As such, the developer is given
+// the option to use alphabetically sorted json w/ nlohmann::json or insertion
+// order preserved json with nlohmann::ordered_json.
+// Since alphabetical order doesn't help a backend system and insertion order
+// can have benefits when reading user defined json (for example the order of
+// widgets when reading tasks json for creating settings dialogs)
+// we will use nlohmann::ordered_json to preserve insertion order.
+using json = nlohmann::ordered_json;
+
 namespace FOEDAG {
 
 class Settings {
  private:
-  QJsonObject m_data;
+  json m_json;
 
  public:
   Settings();
-  void setValue(QJsonObject& parent, const QString& key,
-                const QJsonValue& value);
-  void setValue(const QString& key, const QJsonValue& value);
-  void setString(const QString& key, const QString& string);
-  void updateJson(const QString& key, QJsonObject& newJson);
-  void updateJson(QJsonObject& parent, const QString& key,
-                  QJsonObject& newJson);
-
-  QJsonValue get(QJsonObject& parent, const QString& key);
-  QJsonValue get(const QString& key);
-  QJsonValue getNested(QJsonObject& parent, const QString& keyPath,
-                       const QString& pathSeparator = ".");
-  QJsonValue getNested(const QString& keyPath,
-                       const QString& pathSeparator = ".");
-  QString getJsonStr(const QJsonObject& object);
+  virtual void loadSettings();
```

Comment:
why is virtual? I don't think it is make sense to have some class hierarchy for settings.